### PR TITLE
Parse trulia's bullet items to add meaningful keys

### PR DESCRIPTION
**Pull request by algodave**
@jeffdeville 

You see on the Trulia page we have 2 bulleted lists: a summary and a "Public Records" section.
I was scraping those lists as array of strings, but when it comes to fitting those strings into a domain model, I need to have meaningful keys to understand what those values are.

### CODE REVIEW

**Comment by algodave** [comment]
Sauce Labs authentication is failing on Travis because @jeffdeville 's credentials can't work on algonauti's repo. This policy turns out to be really annoying... 
However, I was able to run those test successfully on my local - interestingly, that policy does not apply in that case - and they were all green, so @jeffdeville you can safely merge to master

File: lib/sherlock_homes/trulia.rb [hunk 1]
```diff
@@ -64,27 +64,62 @@ def parse
...
+        result = Hash.new
         @data.css('li').each do |group|
           group.css('ul > li').each do |li|
-            result << li.text.strip
+            result.merge! BulletItemParser.parse(li.text.strip)
           end
         end
         result
       end
     end
```
Comment (jeffdeville) [inline lib/sherlock_homes/trulia.rb]: What is AbstractParser buying us here?  Seems like:
```ruby
module BulletItemParser
  def self.parse(data)
    # do things
  end
end

would be sufficient. Or if you wanted to use lambdas:

```ruby
    ImageUrlParser = lambda do |data|
      element = data.first
      style = element.attributes['style'].value
      /background-image:url\('(.*)'\);/.match(style)[1]
    end
  ImageUrlParser.call(data)
```

**Comment by jeffdeville** [comment]
:+1: Feel free to merge if you're comfortable with the specs passing, @algodave

File: lib/sherlock_homes/trulia.rb (hunk 1 above)
Comment (algodave) [inline lib/sherlock_homes/trulia.rb]: @jeffdeville Thanks you for the suggestion, I'm going to define those parsers as lambdas. Sometimes I still suffer a strict OOP mindset, whereas I should think more functional!

**Comment by algodave** [comment]
@jeffdeville Thank you for your trust :+1: but currently I don't have write access to this repo so I can't merge.

Consider granting write access to both me and @safeforge so that we can branch directly on this repo rather than having to do it on our fork. That way you would also see Travis builds green on our PRs.

**Pull request merged**